src/cloud_auth.ts

```diff
@@ -7,51 +7,69 @@ import { User } from './config_types';
 
 export class CloudAuth implements Authenticator {
     public isAuthProvider(user: User): boolean {
-        return user.authProvider.name === 'azure' ||
-            user.authProvider.name === 'gcp';
+        return (
+            user.authProvider.name === 'azure' ||
+            user.authProvider.name === 'gcp'
+        );
     }
 
     public getToken(user: User): string | null {
         const config = user.authProvider.config;
-        // This should probably be extracted as auth-provider specific plugins...
-        let token: string = 'Bearer ' + config['access-token'];
+        if (this.isExpired(config)) {
+            this.updateAccessToken(config);
+        }
+        return 'Bearer ' + config['access-token'];
+    }
+
+    private isExpired(config) {
```

Comment:
Add type information here.

Reply:
agree, but there is no type definition for `authProvider`, it's `any`-type, so i would have to invent a new type for the config.